zrml/swaps/src/utils.rs

```diff
@@ -157,56 +157,63 @@ where
 
     let [asset_amount_in, asset_amount_out] = (p.asset_amounts)()?;
 
-    if p.pool.scoring_rule == ScoringRule::CPMM {
-        T::Shares::transfer(p.asset_in, &p.who, p.pool_account_id, asset_amount_in)?;
-        T::Shares::transfer(p.asset_out, p.pool_account_id, &p.who, asset_amount_out)?;
-    } else {
-        let base_asset = p.pool.base_asset;
-
-        if p.asset_in == base_asset {
+    match p.pool.scoring_rule {
+        ScoringRule::CPMM => {
             T::Shares::transfer(p.asset_in, &p.who, p.pool_account_id, asset_amount_in)?;
-            T::Shares::deposit(p.asset_out, &p.who, asset_amount_out)?;
-        } else if p.asset_out == base_asset {
-            // We can use the lightweight withdraw here, since event assets are not reserved.
-            T::Shares::withdraw(p.asset_in, &p.who, asset_amount_in)?;
             T::Shares::transfer(p.asset_out, p.pool_account_id, &p.who, asset_amount_out)?;
-        } else {
-            // Just for safety, should already be checked in p.asset_amounts.
-            return Err(Error::<T>::UnsupportedTrade.into());
+        }
+        ScoringRule::RikiddoSigmoidFeeMarketEma => {
+            let base_asset = p.pool.base_asset;
+
+            if p.asset_in == base_asset {
+                T::Shares::transfer(p.asset_in, &p.who, p.pool_account_id, asset_amount_in)?;
+                T::Shares::deposit(p.asset_out, &p.who, asset_amount_out)?;
+            } else if p.asset_out == base_asset {
+                // We can use the lightweight withdraw here, since event assets are not reserved.
+                T::Shares::withdraw(p.asset_in, &p.who, asset_amount_in)?;
+                T::Shares::transfer(p.asset_out, p.pool_account_id, &p.who, asset_amount_out)?;
+            } else {
+                // Just for safety, should already be checked in p.asset_amounts.
+                return Err(Error::<T>::UnsupportedTrade.into());
+            }
         }
     }
 
     let spot_price_after = Pallet::<T>::get_spot_price(p.pool_id, p.asset_in, p.asset_out)?;
 
     // Allow little tolerance
-    if p.pool.scoring_rule == ScoringRule::RikiddoSigmoidFeeMarketEma {
-        ensure!(
-            spot_price_before.saturating_sub(spot_price_after) < 20u8.into(),
-            Error::<T>::MathApproximation
-        );
-    } else {
-        ensure!(spot_price_after >= spot_price_before, Error::<T>::MathApproximation);
+    match p.pool.scoring_rule {
+        ScoringRule::CPMM => {
+            ensure!(spot_price_after >= spot_price_before, Error::<T>::MathApproximation);
+        }
+        ScoringRule::RikiddoSigmoidFeeMarketEma => {
+            ensure!(
+                spot_price_before.saturating_sub(spot_price_after) < 20u8.into(),
+                Error::<T>::MathApproximation
+            );
+        }
```

Comment:
If we're going with the Parity style guide, these blocks are unnecessary.

Reply:
`cargo fmt` wants to use blocks here somehow. I would probably stick with `cargo fmt` here, because when not using it, the workflow fails.